### ISSUE TITLE
GraphQL client (default) is missing it's host.

### ISSUE THREAD

**Issue by Saurou** [Diizzayy/nuxt-graphql-client#321]
### Environment

Nuxt 3

### Describe the bug

`WARN GraphQL client (default) is missing it's host.` is shown after running yarn on server.

It also has a typo, as it shoulod be `its` instead of `it's`.

### Expected behaviour

host is added to nuxt.config.ts

```
export default defineNuxtConfig({
  'graphql-client': {
    clients: {
      default: {
        host: process.env.GQL_HOST,
        token: process.env.TOKEN,
        retainToken: true
      }
    }
  }
}
```

### Additional context

VS Code also marks `host:` as an error with the following description:
```
Type 'string | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string’.
```

**Comment by Diizzayy** [comment]
@Saurou The configuration you have above is correct and should in fact be working properly, have you ensured that the `GQL_HOST` environment variable is set?


> VS Code also marks host: as an error with the following description:
> ```
>Type 'string | undefined' is not assignable to type 'string'.
> Type 'undefined' is not assignable to type 'string’.
>```

the `host` property expects a string, the keys within `process.env` have a type of `string | undefined`, hence the type error.

Also, this module is configured to automatically detect the `GQL_HOST` variable, so you don't have to explicity set `{ host: process.env.GQL_HOST }`

**Comment by Saurou** [comment]
@Diizzayy thanks.

It looks like the .env was deleted on the server, for some reason, hence why the error. That's also why I was not getting the warning locally.

I removed `host` from nuxt.config.ts settings, thanks for the suggestion. VS Code does however still mark both `host` and `token` with that error message, even if they both are declared on my local .env.

**Comment by bastianhilton** [comment]
I'm having the same issue even though my .env has the GQL_HOST='http://example.com' declared and no host in the GQL plugin.

**Comment by ajaxbo360** [comment]
@Saurou yup i also get the same error when deploying in my server ,however in my local its working fine without the host ,env on production like was deleted wierd, this you manage to fix the error on deployment

**Comment by ajaxbo360** [comment]
@Diizzayy hope you doing. great i just wanna mention that this bug still not be fixed especially when deploying to server i get the error above the host is missing , i did the approch you give and yes it works on local environment but in production no you can’t use process.env.GQL_HOST ,HOPE YOU GUYS CAN FIX THIS ASAP

**Comment by Diizzayy** [comment]
@ajaxbo360 Thanks for resurfacing this issue. I'll be sure to investigate this today

**Comment by ajaxbo360** [comment]
@Diizzayy thanks bro much appreciated ,basically the env variable is not been used in host ,cause i tried on local ,everything is working i commented host but in production still give me the same issue host is missing, hope you have a wonderful day

**Comment by ajaxbo360** [comment]
@Diizzayy i did some investigation ,this where the problem came from in production it wont work it still gives me the error:  "graphql-client": {
    watch: true,
    clients: {
      default: {
        host:"http://example:3001/graphql",
        corsOptions: {
          credentials: "include",
        },
      },
    },
    autoImport: true,
    functionPrefix: "Gql",
    documentPaths: ["./"],
    preferGETQueries: false,
  }, and i noticed in my last project i didn’t use cookie soo when you try to remove "graphql-client": {
    watch: true,
    clients: {
      default: {
        host:"http://example:3001/graphql",
        corsOptions: {
          credentials: "include",
        },
      },
    },
    autoImport: true,
    functionPrefix: "Gql",
    documentPaths: ["./"],
    preferGETQueries: false,
  },it stills gives me the error.

**Comment by Tom-El-Egant** [comment]
Facing the same issue.
I discovered that the env variable has to be set on build time - changing it on runtime does not have any effect.
Since we deploy our nuxt application together with graphQL application in different environments with different urls, we only know the GQL_HOST on runtime (same dockerfile for build, but on runtime different hosts).
Any suggestions?
